app/src/main/java/org/sagebionetworks/research/mpower/room/RoomScheduledActivity.kt

```diff
@@ -0,0 +1,130 @@
+package org.sagebionetworks.research.mpower.room
+
+import android.arch.persistence.room.ColumnInfo
+import android.arch.persistence.room.Embedded
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Index
+import android.arch.persistence.room.PrimaryKey
+import com.google.gson.annotations.SerializedName
+import org.joda.time.DateTime
+import org.sagebionetworks.bridge.rest.model.ActivityType
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+
+import java.util.ArrayList
+
+@Entity
+data class RoomScheduledActivity(@SerializedName("guid") @PrimaryKey var guid: String) {
+    @SerializedName("schedulePlanGuid")
+    var schedulePlanGuid: String? = null
+
+    @SerializedName("startedOn")
+    var startedOn: DateTime? = null
+
+    @SerializedName("finishedOn")
+    @ColumnInfo(index = true)
+    var finishedOn: DateTime? = null
+
+    @SerializedName("scheduledOn")
+    @ColumnInfo(index = true)
+    var scheduledOn: DateTime? = null
+
+    @SerializedName("expiresOn")
+    @ColumnInfo(index = true)
+    var expiresOn: DateTime? = null
+
+    @SerializedName("activity")
+    @Embedded(prefix = "activity_")
+    var activity: RoomActivity? = null
+
+    @SerializedName("persistent")
+    var persistent: Boolean? = null
+
+    /// TODO: mdephillips 8/20/18 how do we store a weak type?
+    // One way we could do it, is create our own class type "ClientData" which stores "var data: Any?"
+    // and it will have a type converter to and from String that is dynamic based on type of data: Any.
+    // Thoughts?
+//    @SerializedName("clientData")
+//    var clientData: Any? = null
```

Comment:
a blob? and just write the JSON?

Reply:
And then I'd imagine a specific task would be tied to a POJO which deserializes/serializes clientData

Reply:
I updated the PR to include client data.  I ended up just doing an object wrapper, it works for a weak type storage.